src/proto/h1/io.rs

```diff
@@ -11,15 +11,15 @@ use tokio_io::{AsyncRead, AsyncWrite};
 use super::{Http1Transaction, ParseContext, ParsedMessage};
 
 /// The initial buffer size allocated before trying to read from IO.
-pub(crate) const INIT_BUFFER_SIZE: usize = 8192;
+pub(crate) const INIT_BUFFER_SIZE: usize = 64 * 1024;
 
 /// The minimum value that can be set to max buffer size.
 pub const MINIMUM_MAX_BUFFER_SIZE: usize = INIT_BUFFER_SIZE;
 
 /// The default maximum read buffer size. If the buffer gets this big and
 /// a message is still not complete, a `TooLarge` error is triggered.
 // Note: if this changes, update server::conn::Http::max_buf_size docs.
-pub(crate) const DEFAULT_MAX_BUFFER_SIZE: usize = 8192 + 4096 * 100;
+pub(crate) const DEFAULT_MAX_BUFFER_SIZE: usize = 8192 * 1024;
```

Comment:
Ok.  But we'll need to somehow let the user know that if they set the read_buf/chunk_size they'll need to also up the max buffer size otherwise it will panic.  

My inclination is to get rid of the initial check, and assume that if the user ups the read_buf_size they know what they are doing..  I'm not sure, though.

Reply:
thinking about this a bit more, I think maybe we just set the max size to 4x the initial buffer size if the user updates the initial buffer size.

Reply:
Setting the max buffer size is currently only done on the server side (via `Http::max_buf_size`). I think for this new API, this should be about setting an *exact* (as in, always use exactly and only this size) read buffer size. Thus, (assuming `read_buf_exact_size` as the Builder function name), calling `Conn::set_read_buf_exact_size` should also just set `max_buf_size` to the same. WDYT?

Reply:
If we take this approach,  I *think* that we'll need to go to a two buffer system,  since the read buffer is extended as needed to handle the contents of the `AsyncRead::read`.  I am fairly sure that, at this function, the buffer size is compared to the max buf size, even on the client.

Reply:
IOW,  we have a single, contiguous buffer that gets extended to handle the entire http message (if it is not chunked)